pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,206 @@
+  if [[ -z "$repo_exists_status" ]]; then
+    if kubectl get "cronjob" -n "opensearch-system" "opensearch-backup" &> /dev/null
+    then
+        last_schedule=$(kubectl get cronjob -n opensearch-system opensearch-backup -ojsonpath='{.status.lastScheduleTime}')
+        if [[ $last_schedule != `date +"%Y-%m-%dT12:00:00Z"` ]]; then
+          echo -e "failure ❌";
+          echo "[ERROR] opensearch-backup cronjob has not been scheduled"
+          return 1
+        fi
+        echo -e "success ✔";
+        return 0
+    else
+        echo -e "failure ❌";
+        echo "[ERROR] opensearch-backup cronjob doesn't exist"
+        return 1
+    fi
+  else
+    echo -ne "failure ❌";
+    echo -e $repo_exists_status | jq
+  fi
+}
+
+check_opensearch_indices(){
+  echo -ne "Checking opensearch indices status ... "
+
+  for index in 'other' 'kubernetes' 'kubeaudit' 'authlog'; do
+    res=$(curl -w "%{http_code}"  -o /dev/null -ksIL -u admin:${adminPassword} -X HEAD "https://opensearch.${opsDomain}/${index}")
+    if [[ $res != "200" ]]; then
+      echo "failure ❌";
+      echo "[ERROR] Missing index : ${index}"
+      return 1
+    fi
+  done
+
+  echo "success ✔"
+  return 0
+}
+
+check_opensearch_breakers(){
+  echo -ne "Checking opensearch breakers ... "
+  breakers=$(curl -sk -u admin:${adminPassword} -X GET "https://opensearch.${opsDomain}/_nodes/_all/stats/breaker")
+
+  if [[ $(echo $breakers | jq '.nodes[].breakers | has("fielddasa")') == "true" ]] && \
+    [[ $(echo $breakers | jq '.nodes[].breakers | has("in_flight_requests")') == "true"  ]] && \
+    [[ $(echo $breakers | jq '.nodes[].breakers | has("parent")') == "true"  ]] && \
+    [[ $(echo $breakers | jq '.nodes[].breakers | has("request")') == "true"  ]]; then
+
+      echo -e "success ✔";
+      return 0
+  else
+    echo -e "failure ❌";
+    echo "[ERROR] There is a missing breaker (fielddata, parent, in_flight_requests, request)"
+    echo $breakers | jq
+    return 1
+  fi
+
+
+}
+
+check_opensearch_aliases(){
+  echo -ne "Checking opensearch aliases & indices ... "
+  curl -sk -o /tmp/response -u admin:${adminPassword} -X GET "https://opensearch.${opsDomain}/_cat/aliases"
+
+  aliases=$(cat /tmp/response | awk '{print $1}')
+  indices=$(cat /tmp/response | awk '{print $2}')
+  aliases_arr=($aliases)
+  indices_arr=($indices)
+  len=${#aliases_arr[@]}
+  ((len--))
+
+  for i in  `seq 0 $len`; do
+    if [[ ${aliases_arr[$i]} =~ ^(other|kubernetes|kubeaudit|authlog)$ ]]; then
+
+      if ! grep -q ${aliases_arr[$i]} <<< ${indices_arr[$i]}; then
+        echo "failure ❌";
+        echo "[ERROR] Incorrect Alias ${aliases_arr[$i]} for Index ${indices_arr[$i]}"
+        return 1
+      fi
+    fi
+  done
+
+  echo "success ✔"
+  return 0
+
+}
+
+check_opensearch_mappings(){
+  curl -sk -u admin:${adminPassword} -X GET "https://opensearch.${opsDomain}/_mapping" | jq
+}
```

Comment:
The thing we want from this is to check so that the number of mappings/fields per index is in a healthy amount compared to the field limit configured on the indices themselves.

Your can use get on `<opensearch-url>/<index>/_field_caps?fields=*` to get a nice output of all mapped fields. And if you get the settings of the index you'll get the limit.

Reply:
Thanks a lot, I was having a hard time finding what to test on this